### PR TITLE
Asdf update command

### PR DESCRIPTION
**Pull request by Stratus3D**
This should complete #172.

### CODE REVIEW

**Comment by Stratus3D** [comment]
Updated this PR with tests for the update command.

File: lib/commands/update.sh [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    tag=$(echo $tags | cut -d ' '  -f1)
+
+    # Update
+    git checkout "$tag" || exit 1
+    echo "Updated asdf to release $tag"
+  fi
+  )
+}
+
+# stolen from https://github.com/rbenv/ruby-build/pull/631/files#diff-fdcfb8a18714b33b07529b7d02b54f1dR942
```
Comment (HashNuke) [inline lib/commands/update.sh]: Haha. Thanks for adding appropriate credits :)

**Comment by HashNuke** [comment]
Build breaks for some reason.

File: lib/commands/update.sh (hunk 1 above)
Comment (Stratus3D) [inline lib/commands/update.sh]: I actually stole this from the irdris plugin, which stole it from ruby build :p https://github.com/vic/asdf-idris/blob/master/bin/list-all#L6

**Comment by Stratus3D** [comment]
Looks like `tail -r` to get the last item doesn't work on Linux. I switched to a simple sed find and replace expression to get the latest tag. We are already using sed in several places so it's guaranteed to work on any OS that asdf currently works on.

**Comment by Stratus3D** [comment]
That fix was completely wrong. I've pushed the corrected the sed command.

**Comment by Stratus3D** [comment]
This passes on both my Ubuntu laptop and my OSX laptop. I also ran the tests on my CentOS 6 server and tests passed there too. Not sure what is causing the tests to fail on Travis.

I tried using Vagrant but ran into issues, I'll see if I can get them sorted out tomorrow.

**Pull request closed without merging**